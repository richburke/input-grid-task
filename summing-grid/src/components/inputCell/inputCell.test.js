import React from 'react';
import { shallow } from 'enzyme';
import InputCell from './';

describe('InputCell', () => {
  it('should render correctly', () => {
    const component = shallow(<InputCell />);
    expect(component).toMatchSnapshot();
  });

  it('should contain an input', () => {
    const component = shallow(<InputCell />);
    expect(component.find('input')).toBeDefined();
  });

  it('should pass the value of the input to the onChange handler', () => {
    const onChangeMock = jest.fn();

    const component = shallow(
      <InputCell
        onChange={onChangeMock}
        rowIndex={1}
        columnIndex={5}
      />
    );
    component.find('input').simulate('change', { target: { value: '12.3' } });
    expect(onChangeMock).toBeCalledWith({ target: { value: '12.3'} }, 1, 5);
  });
});
