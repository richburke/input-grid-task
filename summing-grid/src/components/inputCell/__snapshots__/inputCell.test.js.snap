// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCell should render correctly 1`] = `
<Cell>
  <input
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
</Cell>
`;
